### ISSUE TITLE
It would be usefull to allow using the config or a constructor to pass a test console which could be configured externally

### ISSUE THREAD

**Issue by scarbon** [spectreconsole/spectre.console#1718]
https://github.com/spectreconsole/spectre.console/blob/58bf89a56a7f747665acff45484abf9182fd3977/src/Spectre.Console.Testing/Cli/CommandAppTester.cs#L107

It would be usefull to allow using the config or a constructor to pass a test console which could be configured externally, I would like to do integration test with interactive test console but cant set iy up here

**Comment by FrankRay78** [comment]
Related to https://github.com/spectreconsole/spectre.console/issues/1737

**Comment by FrankRay78** [comment]
Let's get this PR motoring @scarbon.

**Comment by byte2pixel** [comment]
> Related to [#1737](https://github.com/spectreconsole/spectre.console/issues/1737)
Yep exactly what I was thinking with changes I was looking for when Iwas doing advent of code 

Dang I was totally going to make this PR I was waiting for reply to my issue. That it was ok

**Comment by magiino** [comment]
Done. Please review the PR and let me know your thoughts! Thanks
